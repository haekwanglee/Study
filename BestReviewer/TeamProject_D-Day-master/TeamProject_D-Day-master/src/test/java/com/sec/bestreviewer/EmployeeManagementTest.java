package com.sec.bestreviewer;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;


import java.io.File;
import java.io.FileOutputStream;
import java.io.OutputStream;
import java.text.Format;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class EmployeeManagementTest {
    @Test
    public void testArgumentsEmptyInputFile() throws Exception {
        String[] args = {"./src/test/java/com/sec/bestreviewer/empty.txt", "./src/test/java/com/sec/bestreviewer/output.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();
        employeeManagement.run(args);

        File file = new File("./src/test/java/com/sec/bestreviewer/output.txt");

        assertTrue(file.exists());
    }

    @Test
    public void testArgumentsWrongArgsCount() throws Exception {
        String[] args = {"input.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();

        Assertions.assertThrows(IllegalArgumentException.class, () -> {
            employeeManagement.run(args);
        });
    }

    @Test
    public void testArgumentsNotExistInputFile() throws Exception {
        String[] args = {"notexist.txt", "output.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();

        Assertions.assertThrows(IllegalArgumentException.class, () -> {
            employeeManagement.run(args);
        });
    }

    @Test
    public void integrationTest() throws Exception {
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_output.txt";
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_input.txt";

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    public void integrationModifyTest() throws Exception {
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_modify_output.txt";
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_modify_input.txt";

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }
}
