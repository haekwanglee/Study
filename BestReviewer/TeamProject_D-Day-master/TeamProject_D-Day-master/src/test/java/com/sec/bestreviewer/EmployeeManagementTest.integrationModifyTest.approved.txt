MOD,10
MOD,NONE
MOD,NONE
MOD,18050301,KYUMOK KIM,CL2,010-9777-5455,19980906,ADV
MOD,18050302,KYUMOK KIM,CL2,010-9777-5455,19980906,ADV
MOD,18050303,KYUMOK KIM,CL2,010-9777-5455,19980906,ADV
MOD,18050304,KYUMOK KIM,CL2,010-9777-5455,19980906,PRO
MOD,18050305,KYUMOK KIM,CL2,010-9777-5455,19980906,PRO
MOD,9
MOD,02712462,UAFBOWU HONG,CL3,010-9054-6560,19780825,ADV
MOD,11
